src/main/java/org/apache/sysds/runtime/matrix/data/LibCommonsMath.java

```diff
@@ -209,6 +218,339 @@ private static MatrixBlock computeSolve(MatrixBlock in1, MatrixBlock in2) {
 		
 		return DataConverter.convertToMatrixBlock(solutionMatrix);
 	}
+
+	/**
+	 * Computes the eigen decomposition of a symmetric matrix.
+	*
+	* @param in The input matrix to compute the eigen decomposition on.
+	* @param threads The number of threads to use for computation.
+	* @return An array of MatrixBlock objects containing the real eigen values and eigen vectors.
+	*/
+	public static MatrixBlock[] computeEigenDecompositionSymm(MatrixBlock in, int threads) {
```

Comment:
i would like to know where the time is spend inside this function call.

is it in transformToTridiagonal, getQ or findEigenVectors?

Reply:
![image](https://github.com/apache/systemds/assets/18759490/0013632a-2db5-4e4b-a21f-2738ae1c7fd1)

`findEigenVectors` seems to take ~60-70% of the time.
That performEigenDecomposition is a wrapper for the do-while loop and most of its execution is taken by the matrix updates at the end of each iteration

Reply:
great this is something that is easy to fix, simply do not use the get and set method for the dense block, but directly on the underlying linearized array assign the cells.

This should immediately reduce the execution time by at least ~50% of that part.